src/libsync/propagateuploadng.cpp

```diff
@@ -30,58 +30,213 @@
+           +                                                       +                                       |
+           |no                                                     |                                       |
+           |                                                       |                             Upload .zsync chunk
+           v                                                       |                                       |
+        Check the db: is there an entry? <-------------------------+                                       |
+           +                           +                                                                   |
+           |no                         |yes                                                                |
+           |                           v                                                                   |
+           v                        PROPFIND                                                               |
+           startNewUpload() <-+        +-------------------------------------+                             |
+              +               |        +                                     |                             |
+             MKCOL            + slotPropfindFinishedWithError()     slotPropfindFinished()                 |
+              +                                                       Is there stale files to remove?      |
+          slotMkColFinished()                                         +                      +             |
+              +                                                       no                    yes            |
+              |                                                       +                      +             |
+              |                                                       |                  DeleteJob         |
+              |                                                       |                      +             |
+        +-----+^------------------------------------------------------+^--+  slotDeleteJobFinished()       |
+        |                                                                                                  |
+        |                             +--------------------------------------------------------------------+
+        |                             v
+        +---->  startNextChunk() +-> finished?  +-
+                      ^               +          |
+                      +---------------+          |
+                                                 |
+        +----------------------------------------+
+        |
+        +-> MOVE +-----> moveJobFinished() +--> finalize()
  */
 
 void PropagateUploadFileNG::doStartUpload()
 {
     propagator()->_activeJobList.append(this);
 
+    _zsyncSupported = isZsyncPropagationEnabled(propagator(), _item);
+    if (_zsyncSupported && _item->_remotePerm.hasPermission(RemotePermissions::HasZSyncMetadata)) {
```

Comment:
later: In the case where this is false, one can still have `_zsyncSupported` be true and use the corresponding code paths below. I haven't looked in detail whether that's correct.